grobid-core/src/main/java/org/grobid/core/jni/JEPThreadPool.java

```diff
@@ -64,40 +111,20 @@ private JEPThreadPool() {
      * (or create one the first time). 
      */
     public Jep getJEPInstance() {
-        if (jepInstances.get(Thread.currentThread().getId()) == null) {
-            JepConfig config = new JepConfig();
-            try {
-                File delftPath = new File(GrobidProperties.getInstance().getDeLFTFilePath());
-                if (!delftPath.exists()) {
-                    throw new GrobidResourceException("DeLFT installation path does not exist");
-                }
-                if (!delftPath.isDirectory()) {
-                    throw new GrobidResourceException("DeLFT installation path is not a directory");
-                }
-                config.addIncludePaths(delftPath.getAbsolutePath());
-                config.setClassLoader(Thread.currentThread().getContextClassLoader());
-                //System.out.println("jep instance thread: " + Thread.currentThread().getId());
-                try(Jep jep = new Jep(config)) {
-                    jepInstances.put(Thread.currentThread().getId(), jep);
-                    // import packages
-                    jep.eval("import os");
-                    jep.eval("import numpy as np");
-                    jep.eval("import keras.backend as K");
-                    jep.eval("os.chdir('" + delftPath.getAbsolutePath() + "')");
-                    jep.eval("from delft.utilities.Embeddings import Embeddings");
-                    jep.eval("import delft.sequenceLabelling");
-                    jep.eval("from delft.sequenceLabelling import Sequence");
-                    jep.eval("from delft.sequenceLabelling.reader import load_data_and_labels_crf_file");
-                    jep.eval("from delft.sequenceLabelling.reader import load_data_crf_string");
-                    jep.eval("from sklearn.model_selection import train_test_split");
-                }
-            } catch(JepException e) {
-                LOGGER.error("JEP initialization failed", e);
-            } catch(GrobidResourceException e) {
-                LOGGER.error("DeLFT installation path invalid, JEP initialization failed", e);
-            }
+        long threadId = Thread.currentThread().getId();
+        Jep jep = jepInstances.get(threadId);
+        if (jep == null) {
+            jep = this.createJEPInstance();
+            jepInstances.put(threadId, jep);
+        }
+        try {
+            jep.isValidThread();
```

Comment:
This is not strictly necessary. It's called whenever you call `jep.eval`. It just makes it fail fast or re-create the instance. Although it shouldn't randomly close.

Reply:
I think there might be cases where the python part crash the JEP instance. Although I don't remember exactly why I put this, maybe just paranoia.